File: taejoong/leetcode/739_DailyTemperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        
        Stack<Integer> stack = new Stack<>();
        int[] result = new int[temperatures.length];

        for(int i = 0; i < temperatures.length; i++) {

            while(!stack.isEmpty() && temperatures[i] > temperatures[stack.peek()]) {
                int tem = stack.pop();
                result[tem] = i - tem;
            }

            stack.push(i);
        }

        return result;
    }
}

File: hosung/leetcode/739. Daily Temperatures.py
class Solution(object):
    def dailyTemperatures(self, temperatures):
        n =len(temperatures)
        st = []
        res = [0]*len(temperatures)
        for i in range(n):
            while st and temperatures[i] > temperatures[st[-1]]:
                idx = st.pop()
                res[idx] = i -idx

            st.append(i)
            
        return res
                
                

        
